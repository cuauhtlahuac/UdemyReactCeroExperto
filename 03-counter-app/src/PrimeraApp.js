import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import CounterApp from './CounterApp';
import { getImagen } from 'base-pruebas/11-async-await';

export default function PrimeraApp({ saludo }) {
	useEffect(() => {
		getImagen();
	}, []);
	return (
		<React.Fragment>
			<CounterApp value={0} />
		</React.Fragment>
	);
}

PrimeraApp.defaultProps = {
	saludo: 'Saludo por default'
};

PrimeraApp.propTypes = {
	saludo: PropTypes.string.isRequired
};
