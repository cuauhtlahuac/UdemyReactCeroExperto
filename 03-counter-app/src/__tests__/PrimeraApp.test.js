import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json'; // Para renderizar el componente en snapshots

import PrimeraApp from 'PrimeraApp';

describe('<PrimeraApp />', () => {
	test('should render <PrimeraApp /> correctly', () => {
		
		const saludo = 'Hola, Soy Goku!';

		const wrapper = shallow(<PrimeraApp saludo={ saludo } />);

		expect( toJson( wrapper )).toMatchSnapshot();
	});
});
