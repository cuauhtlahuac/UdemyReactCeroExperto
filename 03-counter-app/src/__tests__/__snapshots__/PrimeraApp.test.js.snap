// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimeraApp /> should render <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku!
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <CounterApp
    value={0}
  />
</Fragment>
`;
